toolchain/sem_ir/typed_insts.h

```diff
@@ -384,8 +384,10 @@ struct BoolLiteral {
   BoolValue value;
 };
 
-// A bound method, that combines a function with the value to use for its
-// `self` parameter, such as `object.MethodName`.
+// A bound method, that combines a function with the value to use for its `self`
+// parameter, such as `object.MethodName`. It is a callable thing like the
+// `FunctionType` it refers to through the type of the `function_id`
+// instruction, rather than a call to the method itself like `Call`.
```

Comment:
This is taking a few readings for me to understand, like I read "type of the `function_id` instruction" to mean C++ type of `inst` instead of `inst.type_id()`. What do you think of something like:

```suggestion
// For member access such as `object.MethodName`, combines the function with the
// value to use for `self`. This is a callable structure; `Call` will handle the
// argument assignment.
```

Reply:
Yeah, that is an improvement. I did have to do a lot of reading to understand the relationship of what the `function_id` was, and why its type was being used for lookups instead of itself directly, so if we could capture that relationship that might be nice. Maybe as a comment on the function_id then.

Reply:
Done, WDYT?